libs/accounts/rate-limit/src/lib/util.ts

```diff
@@ -35,3 +40,85 @@ export function calculateRetryAfter(now: number, block: BlockRecord) {
 export function sanitizeKeyValue(val: string) {
   return val.replace(':', ' ');
 }
+
+export function getLargestRetryAfter(blocks: BlockStatus[]) {
+  if (blocks.length > 0) {
+    let ban = blocks[0];
+    blocks.forEach((x) => {
+      if (x.retryAfter > ban.retryAfter) {
+        ban = x;
+      }
+    });
+    return ban;
+  }
+  return null;
+}
+
+export function isBlockRecord(obj: any): obj is BlockRecord {
+  const result =
+    typeof obj.action === 'string' &&
+    typeof obj.blockingOn === 'string' &&
+    typeof obj.blockedValue === 'string' &&
+    typeof obj.startTime === 'number' &&
+    typeof obj.duration === 'number' &&
+    typeof obj.reason === 'string' &&
+    typeof obj.attempts === 'number' &&
+    typeof obj.policy === 'string';
+  return result;
+}
+
+export function parseBlockRecord(value: string | null): BlockRecord | null {
+  if (!value) {
+    return null;
+  }
+  const json = JSON.parse(value);
+
+  if (!json) {
+    return null;
+  }
+
+  // policy was not initially present and initially all policies were 'block'
+  if (json.policy == null) {
+    json.policy = 'block';
+  }
+
+  if (isBlockRecord(json)) {
+    return json as BlockRecord;
+  }
+
+  return null;
+}
+
+export function createBlockStatus(now: number, block: BlockRecord) {
+  return {
+    startTime: block.startTime,
+    duration: block.duration,
+    attempt: block.attempts,
+    retryAfter: calculateRetryAfter(now, block),
+    reason: block.reason,
+    action: block.action,
+    blockingOn: block.blockingOn,
+    policy: block.policy,
+  } satisfies BlockStatus;
+}
+
+export function createBlockRecord(
+  now: number,
+  action:string,
+  blockedValue: BlockOn,
+  attempts: number,
+  rule: Rule,
+  usedDefaultRule: boolean
+) {
+  return {
+    action: action,
+    startTime: now,
+    duration: rule.blockDurationInSeconds,
+    blockingOn: rule.blockingOn,
+    blockedValue: blockedValue,
+    reason: 'too-many-attempts',
```

Comment:
small suggestion: I feel like I've seen this string enough times to make it a constant.